lib/file/widgets/file_tile.dart

```diff
@@ -22,22 +22,23 @@ class FileTile extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
-    return FancyCachedBuilder<File>.handleLoading(
-      controller: fileId.controller,
-      builder: (context, file, isFetching) {
+    return EntityBuilder<File>(
+      id: fileId,
+      builder: handleEdgeCases((context, file, _) {
         final subtitle = [
           if (file.isActualFile) file.sizeAsString,
           if (file.updatedAt != null) file.updatedAt.shortDateTimeString,
         ].join(', ');
+        final onTap = onTapHandler(file);
 
         return ListTile(
           title: Text(file.name),
           subtitle: Text(subtitle),
           leading: FileThumbnail(file: file),
-          onTap: onTapHandler(file) == null ? null : () => onTapHandler(file),
+          onTap: onTap == null ? null : () => onTap(file),
```

Comment:
Currying to the rescue!
```suggestion
          onTap: onTap?.partial(file),
```